### PR TITLE
Resolve issue #3, reorganize directory structure, build with CMake, adds Doxygen comments and generation via CMake.

### PR DESCRIPTION
**Pull request by madratman**
Self explanatory title. I won't be surprised if I have messed the `strncpy` part though. Not much experience with C. I don't mind squashing commits and fixing things if you find any errors. 

Reorganized files in `include` and `src`, and used a `CMakeLists.txt` because it is more intuitive to write. And the binaries and the doc being generated in `build` is a good thing 

edit: Alright, the Travis build just failed. :\
edit: Maybe, the cmake version is too high.
edit: oh I need to change the script. And I'll remove the `orig_Makefile`.

### CODE REVIEW

File: src/icmp.c [hunk 1]
```diff
@@ -101,15 +101,13 @@ void send_icmp_packet(int sock_fd, struct icmp_packet *packet_details)
...
-  packet = (char *)malloc(packet_size);
+  packet = (char *)calloc(packet_size, sizeof *packet);
   if (packet == NULL) {
     printf("No memory available\n");
     close_icmp_socket(sock_fd);
     exit(-1);
   }
-  memset(packet, 0, packet_size);
 
-  // Initializing header and payload pointers
```
Comment (madratman) [inline src/icmp.c]: ate up this comment somehow

**Comment by DhavalKapil** [comment]
I'll have a look at your code. Might take some time. College exams starting tomorrow :P

**Comment by DhavalKapil** [comment]
@madratman I've been thinking for a while about adding `cmake`. I've finally decided not to go for it. I want to keep 'icmptunnel' simple.

**Comment by rofl0r** [comment]
thanks for not using cmake <3

**Pull request closed without merging**